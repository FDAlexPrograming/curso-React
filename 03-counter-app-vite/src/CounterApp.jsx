import { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ( { value } ) => {

    // function handleAdd(event){ 
    //     console.log( ' +1 ')
    // } 
    const [ counter, setCounter ] = useState( value );


    const handleAdd = (  ) =>{
        // setCounter( counter + 1);
        // o 
        setCounter( ( c ) => c + 1 );
    }

    const handleSubstract = () => setCounter( counter -1 );
    const handleReset = () => setCounter( value);

    return (
        <>
            <h1> Counter App</h1>
            <h2> { counter } </h2>
        <div className="botones">
            <button onClick={ handleAdd }> Sumar </button>
            <button onClick={ handleSubstract }> Restar </button>
            <button aria-label='btn-reset' onClick={ handleReset }> Resetear </button>

        </div>

        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
} 

