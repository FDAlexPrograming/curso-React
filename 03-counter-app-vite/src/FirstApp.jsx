// // import { Fragment } from "react"

// const getResult = (a, b) => {
//     return a + b;
// }

// export const FirstApp = () => {

//     // const newMessage = {
//         //     title: 'Alex',
//         //     message: 'Hola mundo',
//         // };

//   return (
//     <>
//         <h1> { getResult( 4, 4 ) } </h1>
//         {/* <code> { JSON.stringify( newMessage )  } </code> */}
//         <p>soy un párrafo </p>
//     </>
//   )
// }

import PropTypes from 'prop-types';
export const FirstApp = ( { title, párrafo, name } ) => {
  
  // console.log( props );
  
  
  return (
    <>
        <h1 data-testid="test-title">{ title }</h1>
        <p> { párrafo }</p>
        <p> { párrafo }</p>
        <p> { name }</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  párrafo: PropTypes.string.isRequired,
}

FirstApp.defaultProps = {
//title: ' No hay párrafo ',
  párrafo: ' No hay párrafo ',
  name: ' Alex Farias '
}

