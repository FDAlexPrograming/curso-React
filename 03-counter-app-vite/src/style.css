html, body {
    background-color: #21232A;
    color: rgb(12, 223, 170);
    font-family: 'Helvetica, Arial, sans-serif';
    font-size: 1.3rem;
    padding: 70px ;
}

button {
    padding: 15px;
    border-radius: 100%;
    border: none;
    transition: all 1s;
    margin: .5rem;
}

button:hover{
    color: rgb(17, 187, 187);
    background-color: #000000;
    
}
.botones{
    display: flex;
    padding-right: 1rem;
}