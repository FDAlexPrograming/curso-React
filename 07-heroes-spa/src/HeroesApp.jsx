
export const HeroesApp = () => {
  return (
        <>
            <h1>Horeoes App</h1>
        </>  
   )
}
