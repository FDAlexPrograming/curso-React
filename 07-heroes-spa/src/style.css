* {
    color: #d6d4d4 !important;
    font-family: 'Montserrat', sans-serif;
}
body {
    padding: 20px;
    background:rgb(15 8 18 );
}
nav{
    max-height: 3.5rem;
}

button {
    margin-right: 10px;
   
}

li {
    transition: all .4s;
    background: transparent !important;
    border-radius: .2rem;
    border: none !important;
}

.todoFinalized {
   background: #55cd97c4 !important;
}
a{
    font-weight: bold !important;
    color: #969696 !important;
}

a:hover{
    color: #d6d4d4 !important;
}

li:hover {
   background: #2c3943 !important;
}

input {
    background: #dadada !important;        
}

li {
    margin-left: 1rem  ;
    padding-left: 1rem;
    padding-right: 1rem;
}

.active{
    color: #fff !important;
}

.card-text {
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
}