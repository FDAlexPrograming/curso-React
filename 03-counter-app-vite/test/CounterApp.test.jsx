import { fireEvent, render, screen } from '@testing-library/react';
import { CounterApp } from "../src/CounterApp";


describe('Pruebas en CounterApp', () => {
    const initialValue = 100;
    test('debe de hacer match con el snapshot', () => {
        const { container } = render( <CounterApp value= { initialValue } /> );  
        expect( container ).toMatchSnapshot();
        
    });

    test('debe de mostrar el valor inicial de 100', () => {  
        render( <CounterApp value= { initialValue } /> );
        expect( screen.getByText( 100 )).toBeTruthy();
    });

    test('debe de incrementar con el botón Sumar', () => {  
        render( <CounterApp value= { initialValue } /> );
        fireEvent.click( screen.getByText( 'Sumar') );
        expect( screen.getByText('101') ).toBeTruthy();
    });

    test('debe de decrementar con el botón Restar', () => {  
        render( <CounterApp value= { initialValue } /> );
        fireEvent.click( screen.getByText( 'Restar') );
        expect( screen.getByText('99') ).toBeTruthy();
    });

    test('debe de funcionar el botón Resetear', () => {  
        render( <CounterApp value= { initialValue } /> );
        fireEvent.click( screen.getByText( 'Sumar') );
        fireEvent.click( screen.getByText( 'Sumar') );
        fireEvent.click( screen.getByText( 'Sumar') );
        fireEvent.click( screen.getByRole('button', { name: 'btn-reset'}) );
        expect( screen.getByText('100') ).toBeTruthy();

    });

});