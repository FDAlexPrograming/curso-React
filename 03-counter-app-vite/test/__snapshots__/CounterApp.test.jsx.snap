// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
     Counter App
  </h1>
  <h2>
     
    100
     
  </h2>
  <div
    class="botones"
  >
    <button>
       Sumar 
    </button>
    <button>
       Restar 
    </button>
    <button
      aria-label="btn-reset"
    >
       Resetear 
    </button>
  </div>
</div>
`;
